### ISSUE TITLE
why is model in eval mode in train

### ISSUE THREAD

**Issue by abaqusstudent** [talreiss/Mean-Shifted-Anomaly-Detection#3]
I am wondering as to why the pretrained model in: https://github.com/talreiss/Mean-Shifted-Anomaly-Detection/blob/a02ac3001f96fcfa2f9b6a5fcb58cb3e9ba7f5b0/main.py#L32

is in eval mode? When I see the pytorch tutorials for transfer learning, the eval mode is only used during inference and so I am not sure why the model here is in eval mode? Thinking about it I could not come up with a logical explanation for this looking at the main.py file. I would have thought that in https://github.com/talreiss/Mean-Shifted-Anomaly-Detection/blob/a02ac3001f96fcfa2f9b6a5fcb58cb3e9ba7f5b0/main.py#L46 the model should be in train mode. 

Thank you in advance.

Rachel

**Comment by thearkamitra** [comment]
I am not sure but I think one of the reasons might be to remove the effect of Batch-normalization and Dropout. If it was in train mode, the centers would have been heavily affected by the changes introduced by BN.

This is a logical explanation. The authors can confirm the same.

**Comment by cyh-0** [comment]
But if model.eval() is on, the model’s parameters will not be updated.

**Comment by thearkamitra** [comment]
They do get updated. The nature of some of the layers changes based on eval vs train.
You can train an MNIST with the model in eval mode. If weights were not updating, the loss would not have decreased.

**Comment by talreiss** [comment]
Hi,

Indeed the reason for us training the model on eval mode is to remove the effect of batch norm and dropout layers.
